src/engine/SCons/Platform/win32.py

```diff
@@ -178,61 +178,68 @@ def piped_spawn(sh, escape, cmd, args, env, stdout, stderr):
     #   we redirect it into a temporary file tmpFileStdout
     #   (tmpFileStderr) and copy the contents of this file
     #   to stdout (stderr) given in the argument
+    # Note that because this will paste shell redirection syntax
+    # into the cmdline, we have to call a shell to run the command,
+    # even though that's a bit of a performance hit.
     if not sh:
         sys.stderr.write("scons: Could not find command interpreter, is it in your PATH?\n")
         return 127
-    else:
-        # one temporary file for stdout and stderr
-        tmpFileStdout = os.path.normpath(tempfile.mktemp())
-        tmpFileStderr = os.path.normpath(tempfile.mktemp())
-
-        # check if output is redirected
-        stdoutRedirected = 0
-        stderrRedirected = 0
-        for arg in args:
-            # are there more possibilities to redirect stdout ?
-            if arg.find( ">", 0, 1 ) != -1 or arg.find( "1>", 0, 2 ) != -1:
-                stdoutRedirected = 1
-            # are there more possibilities to redirect stderr ?
-            if arg.find( "2>", 0, 2 ) != -1:
-                stderrRedirected = 1
-
-        # redirect output of non-redirected streams to our tempfiles
-        if stdoutRedirected == 0:
-            args.append(">" + str(tmpFileStdout))
-        if stderrRedirected == 0:
-            args.append("2>" + str(tmpFileStderr))
-
-        # actually do the spawn
+
+    # one temporary file for stdout and stderr
+    tmpFileStdout, tmpFileStdoutName = tempfile.mkstemp(text=True)
+    os.close(tmpFileStdout)  # don't need open until the subproc is done
```

Comment:
Why close the file as soon as it's created and then open it later?

Reply:
It's going to be passed on to the compiler command to redirect into, we're not interested in the results until that step is done.  I suppose it doesn't actually hurt anything to leave it open, but we want (and already did) open it as a file object later anyway, not as an os-level descriptor.  (I have no proof not closing causes any issue, just trying to be cautious)